speech/grpc/transcribe_streaming.py

```diff
@@ -90,19 +90,22 @@ def _audio_data_generator(buff):
         yield b''.join(data)
 
 
-def _fill_buffer(audio_stream, buff, chunk):
+def _fill_buffer(audio_stream, buff, chunk, stoprequest):
     """Continuously collect data from the audio stream, into the buffer."""
     try:
-        while True:
+        while not stoprequest.isSet():
             buff.put(audio_stream.read(chunk))
     except IOError:
         # This happens when the stream is closed. Signal that we're done.
         buff.put(None)
 
+    audio_stream.stop_stream()
+    audio_stream.close()
+
```

Comment:
Ah - this is because `buff.put(None)` is never executed if the above loop is exited because `stoprequest.is_set()` is `True`. (this means the blocking `buff.get()` in `_audio_data_generator` blocks forever, since it's never given the `None` to exit from its loop).

If you change the above loop to:

```py
try:
    ...
except IOError:
    pass
finally:
    buff.put(None)
```

then you should be able to put this back.